### PR TITLE
Ensures two thumbs do not get locked at max value

### LINKED ISSUE #2: Two Thumbs Stuck at Max Value
I noticed an issue using the range slider via baseweb actually, but digging around I realized this component being used under the hood exhibited the same behavior.

When two thumbs are shown and both moved to the maximum value, they are then stuck at that value, as the max slider cannot be moved to the left past the min slider, and it is rendered on top. Example below.

sliders-stuck

### PR DESCRIPTION
**Pull request by benmkramer**
Happy to make changes, but just wanted to get an example fix in a PR as I just opened the issue.

Fixes #2 

fixed-two-sliders

### CODE REVIEW

**Comment by tajo** [comment]
**Good catch and thanks for the PR!** The fix works but there are two improvements that could be made:
1. `react-range` supports multiple (more than two) thumbs, your solution assumes only two thumbs. I would like to see a fix for any number of thumbs.
2. If you drag the thumb very close to the max, it doesn't trigger the `zIndex` rule and then it's still very hard to interact with the first thumb:

Screen Shot 2019-03-18 at 11 18 34 AM

Maybe it should consider some delta instead of `val === MAX`. That could be tricky, a robust solution would do some measurements but it might be good enough to use something as `99.5%` of the value range.

Also, it would be awesome to add an e2e test for this! Thanks!

**Comment by benmkramer** [comment]
Thanks for the feedback, will continue working on this soon!

**Comment by benmkramer** [comment]
I wrote a new fix, and a test.

However, I'm still having problems with one scenario. If a rightmost slider is moved to max, then the next rightmost slider is moved to just below max, e.g. 99.5, it is very hard (seems impossible) to grab the edge of that left slider to move it. I feel like I need some way to calculate a buffer range based on the width of the thumb and figure out a range of values during which my fix should apply (e.g. 98-100, rather than just an exact overlap at 100). 

I'll push my change for reference though. I also updated the multi test, but it only seems to use two thumbs / values, so I might add an additional test for 3 sliders as the same bug can occur.

File: src/Range.tsx [hunk 1]
```diff
@@ -327,6 +327,20 @@ class Range extends React.Component<IProps> {
       disabled
     } = this.props;
     const { draggedThumbIndex } = this.state;
+
+    const zIndexes = new Array(values.length).fill(0);
+    const lenValues = values.length;
+
+    // Only need special handling of zIndexes when there are multiple values
+    // and at least one value (the last one) is at the max value
+    if (lenValues > 1 && values[lenValues - 1] === max && values[lenValues - 2] === max) {
```
Comment (tajo) [inline src/Range.tsx]: It should also check for

```js
 !this.props.allowOverlap
```

if overlap is allowed then there is no need for this functionality

**Comment by tajo** [comment]
> I wrote a new fix, and a test.

Nice! Thanks!

> However, I'm still having problems with one scenario. If a rightmost slider is moved to max, then the next rightmost slider is moved to just below max, e.g. 99.5, it is very hard (seems impossible) to grab the edge of that left slider to move it. I feel like I need some way to calculate a buffer range based on the width of the thumb and figure out a range of values during which my fix should apply (e.g. 98-100, rather than just an exact overlap at 100).

Yea, I described that in my previous comment. You can get refs for thumbs with `this.getThumbs()`, use `thumb.getBoundingClientRect()` to get their widths / heights (should work for the vertical case as well) and do some math since you know all the px dimensions and positions. For example if they overlap by more than 50%, apply the zIndex.

**Comment by benmkramer** [comment]
Great, thanks for the input there. I didn't notice the getThumbs returning the refs. :+1: :+1: Will update as soon as I can.

**Comment by tajo** [comment]
Closing since this was fixed in #11

**Pull request closed without merging**